File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i=0
        j=len(height)-1
        res_f=0
        while i<j:
            res=(j-i)*min(height[i],height[j])
            if res>res_f:
                res_f=res
            if height[j]<height[i]:
                j-=1
            else:
                i+=1
        return res_f
        

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i=0;
        int j=height.length-1;
        int res=0;
        int res_m=0;
        while(i<j){
            res_m=(j-i)*(Math.min(height[i],height[j]));
            if(res_m>res){
                res=res_m;
            }
            if(height[j]<height[i]){
                j-=1;
            }
            else{
                i+=1;
            }

        }
        return res;
    }
}